File: evals/code-intelligence/fixtures/batch-c/java/src/main/java/com/acme/service/ItemService.java
package com.acme.service;

import com.acme.model.Item;

interface ItemLoader {
    Item find(String id);
}

public final class ItemService implements ItemLoader {
    public Item find(String id) {
        return new Item(id);
    }

    public Item load(String id) {
        return find(id);
    }

    public Item load(long id) {
        return find(Long.toString(id));
    }
}

File: evals/code-intelligence/fixtures/batch-c/kotlin/src/main/kotlin/com/acme/service/ItemService.kt
package com.acme.service

import com.acme.model.Item

interface ItemLoader {
    fun find(id: String): Item
}

class ItemService : ItemLoader {
    override fun find(id: String): Item = Item(id)

    fun load(id: String): Item = find(id)

    fun load(id: Long): Item = find(id.toString())
}

fun Item.summary(): String = id

fun describe(item: Item): String = item.summary()
